pkg/services/object/get/assembly_v2.go

```diff
@@ -0,0 +1,178 @@
+package getsvc
+
+import (
+	"errors"
+
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	objectSDK "github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"go.uber.org/zap"
+)
+
+func (exec *execCtx) processV2Split(si *objectSDK.SplitInfo) {
+	if _, set := si.FirstPart(); !set {
+		exec.log.Debug("no first ID found in V2 split")
+		exec.err = errors.New("v2 split without first object ID")
+
+		return
+	}
+
+	linkID, set := si.Link()
+	if set && exec.processV2Link(linkID) {
```

Comment:
`processV2Link` could return `false` on partial success: `writeObjectPayload` and `writeCollectedHeader` are called inside. I doubt we should continue processing when they fail

Reply:
changed retuned bool in one place and added comments everywhere